crates/node/engine/src/state/builder.rs

```diff
@@ -1,34 +1,19 @@
 //! An [`EngineState`] builder.
 
-use crate::{EngineClient, EngineState, SyncStatus, client::EngineClientError};
+use crate::{EngineClient, EngineState, EngineStateBuilderError, SyncStatus};
 use alloy_eips::eip1898::BlockNumberOrTag;
-use thiserror::Error;
 
 use kona_protocol::L2BlockInfo;
 
-/// An error that occurs in the [`EngineStateBuilder`].
-#[derive(Error, Debug)]
-pub enum EngineStateBuilderError {
-    /// A temporary error within the engine.
-    #[error("Temporary engine task error: {0}")]
-    EngineClientError(#[from] EngineClientError),
-    /// Missing unsafe head when building the [`EngineState`].
-    #[error("The unsafe head is required to build the EngineState")]
-    MissingUnsafeHead,
-    /// Missing the finalized head when building the [`EngineState`].
-    #[error("The finalized head is required to build the EngineState")]
-    MissingFinalizedHead,
-    /// Missing the safe head when building the [`EngineState`].
-    #[error("The safe head is required to build the EngineState")]
-    MissingSafeHead,
-}
-
 /// A builder for the [`EngineState`].
 ///
-/// When the [`EngineState`] is first created, only the finalized
-/// block is specified. The `StateBuilder` constructs the
-/// [`EngineState`] by fetching the remaining block info via the
-/// client.
+/// The [`EngineStateBuilder`] constructs the [`EngineState`] by fetching
+/// the remaining block info via the [`EngineClient`].
+///
+/// Notice, the finalized and safe heads do *not* fall back to genesis.
+/// This is because Reth has a special branch for the `op-node` (and now `kona-node`)
+/// that kicks off EL sync if the finalized head is not set (e.g. the
+/// finalized head hash is the zero hash).
```

Comment:
```suggestion
/// Notice, the finalized and safe heads do *not* fall back to genesis.
/// This is because Reth has a [special branch][reth-sync] for the `op-node` (and now `kona-node`)
/// that kicks off EL sync if the finalized head is not set (e.g. the
/// finalized head hash is the zero hash).
///
/// [reth-sync]: https://github.com/paradigmxyz/reth/blob/39305dda1c60b4d51333282f408d8c744c34a206/crates/engine/tree/src/tree/mod.rs#L2045-L2065
```